File: src/BukkitPE/event/inventory/CraftItemEvent.php
<?php
namespace BukkitPE\event\inventory;

use BukkitPE\event\Cancellable;
use BukkitPE\event\Event;
use BukkitPE\inventory\Recipe;
use BukkitPE\item\Item;
use BukkitPE\Player;

class CraftItemEvent extends Event implements Cancellable{
	public static $handlerList = null;

	/** @var Item[] */
	private $input = [];
	/** @var Recipe */
	private $recipe;
	/** @var \BukkitPE\Player */
	private $player;


	/**
	 * @param \BukkitPE\Player $player
	 * @param Item[] $input
	 * @param Recipe $recipe
	 */
	public function __construct(Player $player, array $input, Recipe $recipe){
		$this->player = $player;
		$this->input = $input;
		$this->recipe = $recipe;
	}

	/**
	 * @return Item[]
	 */
	public function getInput(){
		$items = [];
		foreach($items as $i => $item){
			$items[$i] = clone $item;
		}

		return $items;
	}

	/**
	 * @return Recipe
	 */
	public function getRecipe(){
		return $this->recipe;
	}

	/**
	 * @return \pocktemine\Player
	 */
	public function getPlayer(){
		return $this->player;
	}
}

File: src/main/java/net/BukkitPE/event/inventory/CraftItemEvent.java
package net.BukkitPE.event.inventory;

import net.BukkitPE.Player;
import net.BukkitPE.event.Cancellable;
import net.BukkitPE.event.Event;
import net.BukkitPE.event.HandlerList;
import net.BukkitPE.inventory.Recipe;
import net.BukkitPE.item.Item;

/**
 * BukkitPE Project
 */
public class CraftItemEvent extends Event implements Cancellable {

    private static final HandlerList handlers = new HandlerList();
    private final Recipe recipe;
    private final Player player;
    private Item[] input = new Item[0];

    public CraftItemEvent(Player player, Item[] input, Recipe recipe) {
        this.player = player;
        this.input = input;
        this.recipe = recipe;
    }

    public static HandlerList getHandlers() {
        return handlers;
    }

    public Item[] getInput() {
        Item[] items = new Item[this.input.length];
        for (int i = 0; i < this.input.length; i++) {
            items[i] = this.input[i].clone();
        }

        return items;
    }

    public Recipe getRecipe() {
        return recipe;
    }

    public Player getPlayer() {
        return player;
    }
}